setup.py

```diff
@@ -9,8 +9,9 @@
     },
     setup_requires=['setuptools_scm'],
     install_requires=[
-        'PyYAML',
-        'jsonschema',
+        'PyYAML>=5.1.2,<5.5',
+        'jsonschema>=3.0.2,<3.3',
+        'pyrsistent~=0.16.1'        # pin to version 0.16 for py27
```

Comment:
Will [environmental markers](https://www.python.org/dev/peps/pep-0508/#environment-markers) work for this? eg 
```suggestion
        'pyrsistent~=0.16.1';python_version="2.7"
```